decentraland/unity-explorer
# fix: emote gender differentiation for abs and embedded

## What does this PR change?

Fixes #1854 

Two problems were detected:
1. Emotes downloaded through ABs were executing a single clip if it was unisex, however it was not considered that the emote could have different clip assets for each gender.
2. There was no differentiation of the gender on embedded emotes.

The solution:
1. Check if the emote entity has different files for each gender, and assign the references accordingly.
2. Implement the gender differentiation for the embedded emotes and assign the references properly during the initial load.

## How to test the changes?

1. Follow issue steps and check that fistbump, dance and robot emotes are executed accordingly to the gender
2. Compare the explorer-web version and check if there is any other emote that require this fix too
3. Check that on-chain emotes play properly by changing the gender

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
